lib/IDE/CodeCompletionResultType.cpp

```diff
@@ -395,8 +395,7 @@ static TypeRelation calculateTypeRelation(Type Ty, Type ExpectedTy,
 
   // Equality/Conversion of GenericTypeParameterType won't account for
   // requirements – ignore them
-  if (!Ty->hasTypeParameter() && !ExpectedTy->hasTypeParameter() &&
-      !Ty->hasUnboundGenericType() && !ExpectedTy->hasUnboundGenericType()) {
+  if (!Ty->hasTypeParameter() && !ExpectedTy->hasTypeParameter()) {
```

Comment:
Can we also get rid of type parameters at some point too, in favor of the caller mapping the type into the correct environment?

Reply:
Yup I'm leaving that for a future PR